rust-wasm-id-allocator/wasm-id-allocator/src/lib.rs

```diff
@@ -203,6 +203,24 @@ impl IdCompressor {
         }
     }
 
+    // Note: bindgen does not allow returning a reference slice
+    pub fn serialize(&self, include_local_state: bool) -> Vec<u8> {
+        self.compressor.serialize(include_local_state)
+    }
+
+    pub fn deserialize(&mut self, bytes: &[u8]) -> Option<IdCompressor> {
```

Comment:
What JS API does this generate? Do you pass in a Uint8Array and bindgen passes a pointer? That would be ideal.

Reply:
deserialize(bytes) {
        const ptr0 = passArray8ToWasm0(bytes, wasm.__wbindgen_malloc);
        const len0 = WASM_VECTOR_LEN;
        const ret = wasm.idcompressor_deserialize(this.ptr, ptr0, len0);
        return ret === 0 ? undefined : IdCompressor.__wrap(ret);
    }

Ends up calling IdCompressor._wrap(ret), just like the constructor. wrap() returns a pointer:
 static __wrap(ptr) {
        const obj = Object.create(IdCompressor.prototype);
        obj.ptr = ptr;

        return obj;
    }

Reply:
Right, but my question is: what is "bytes"? I'm now guessing a Uint8Array, based on the previous answer.

Reply:
Per this doc, yes, we would be sending a Uint8Array:
https://rustwasm.github.io/wasm-bindgen/reference/types/number-slices.html

> #[wasm_bindgen]
pub fn take_number_slice_by_exclusive_ref(x: &mut [u8]) {}

>[Corresponding JS:] 
take_number_slice_by_exclusive_ref(new Uint8Array(100));